source/NVDAObjects/IAccessible/winword.py

```diff
@@ -369,6 +377,32 @@ def script_previousParagraph(self,gesture):
 		self._caretScriptPostMovedHelper(textInfos.UNIT_PARAGRAPH,gesture,None)
 	script_previousParagraph.resumeSayAllMode = sayAll.CURSOR.CARET
 
+	@script(
+		gestures=(
+			"kb:control+v", "kb:control+x", "kb:control+y", "kb:control+z",
+			"kb:alt+backspace", "kb:backspace", "kb:control+backspace"
+		)
+	)
+	def script_updateBrailleAndReviewPosition(self, gesture: inputCore.InputGesture) -> None:
+		"""Helper script to update braille and review position.
+		Caret event is not always fired when control+v, control+x, control+y
+		or control+z (alt+backspace) is pressed so enqueuing caret event for that.
+		When backspace or control+backspace is pressed and hold down, braille
+		may not always be updated. Allowing braille and review position updates
+		in L{event_caret} seems to fix that problem.
+		"""
+		# Ensuring braille and review position updates are allowed in caret event.
+		self._fromUpdateBrailleAndReviewPosition = True
+		# Speech output when backspace or control+backspace is pressed.
+		if gesture._get_displayName() == "backspace":
+			self.script_caret_backspaceCharacter(gesture)
+		elif gesture._get_displayName() == "ctrl+backspace":
+			self.script_caret_backspaceWord(gesture)
+		else:
+			gesture.send()
+			if not eventHandler.isPendingEvents("caret", self):
+				eventHandler.queueEvent("caret", self)

```

Comment:
why not just call event_caret() here?

I think I'd prefer having this change in a separate pr, since basically a fix for this is now hold back by the other work that is done in this pr. As this fix is for UIA and the other fixes are for Word without UIA, they are pretty much unrelated.

Reply:
Changed.

Yes they could/should be separate pull requests. Likely I noticed UIA update problem (braille is not updated when using UIA and braille is tethered to review) when investigating non-uia and then I decided to include fix for it to same pull request.